src/upconversion.rs

```diff
@@ -0,0 +1,252 @@
+use std::collections::{BTreeMap, HashMap, HashSet};
+
+use crate::glyph::GlyphName;
+use crate::ufo::{Groups, Kerning};
+
+pub fn upconvert_kerning(
```

Comment:
can we get a brief docstring here?


Reading through this I do feel that we're doing more cloning than we need to, but I also don't think it's *really* worth the hassle to optimize right now.

Reply:
I'd do it for cleaner code primarily :)